internals/secrethub/service_aws_init.go

```diff
@@ -0,0 +1,95 @@
+package secrethub
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/errio"
+)
+
+// Errors
+var (
+	ErrInvalidAWSRegion = errMain.Code("invalid_region").Error("invalid AWS region")
+	ErrRoleAlreadyTaken = errMain.Code("role_taken").Error("a service using that IAM role already exists")
+)
+
+// ServiceAWSInitCommand initializes a service for AWS.
+type ServiceAWSInitCommand struct {
+	description string
+	path        api.DirPath
+	kmsKeyID    string
+	role        string
+	region      string
+	permission  api.Permission
+	io          ui.IO
+	newClient   newClientFunc
+}
+
+// NewServiceAWSInitCommand creates a new ServiceAWSInitCommand.
+func NewServiceAWSInitCommand(io ui.IO, newClient newClientFunc) *ServiceAWSInitCommand {
+	return &ServiceAWSInitCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+// Run initializes an AWS service.
+func (cmd *ServiceAWSInitCommand) Run() error {
+	repo := cmd.path.GetRepoPath()
+
+	client, err := cmd.newClient()
+	if err != nil {
+		return err
+	}
+
+	cfg := aws.NewConfig()
+	if cmd.region != "" {
+		_, ok := endpoints.AwsPartition().Regions()[cmd.region]
+		if !ok {
+			return ErrInvalidAWSRegion
+		}
+		cfg = cfg.WithRegion(cmd.region)
+	}
+
+	service, err := client.Services().AWS().Create(repo.Value(), cmd.description, cmd.kmsKeyID, cmd.role, cfg)
+	if err == api.ErrCredentialAlreadyExists {
+		return ErrRoleAlreadyTaken
+	} else if err != nil {
+		return err
+	}
+
+	if cmd.permission != 0 {
+		_, err = client.AccessRules().Set(cmd.path.Value(), cmd.permission.String(), service.ServiceID)
+		if err != nil {
+			_, delErr := client.Services().Delete(service.ServiceID)
+			if delErr != nil {
+				fmt.Fprintf(cmd.io.Stdout(), "Failed to cleanup after creating an access rule for %s failed. Be sure to manually remove the created service account %s: %s\n", service.ServiceID, service.ServiceID, err)
+				return errio.Error(delErr)
+			}
+
+			return errio.Error(err)
+		}
+	}
+
+	fmt.Fprintf(cmd.io.Stdout(), "The service %s is now reachable through AWS when the role %s is assumed.\n", service.ServiceID, cmd.role)
+
+	return nil
+}
+
+// Register registers the command, arguments and flags on the provided Registerer.
+func (cmd *ServiceAWSInitCommand) Register(r Registerer) {
+	clause := r.Command("init", "Create a new AWS service account attached to a repository.")
```

Comment:
What is an AWS service? Let's try to make this self-explanatory, e.g. `Create a new service account that is tied to an AWS IAM role and given access to a repository.`

Reply:
Yes, let's add something about the IAM role :+1: 

What does `given access to a repository` mean? I think that might be confusing, as access rules still have to be set.

Reply:
If we change that, let's change it for `secrethub service init` as well, as we also use `attach` there.

Reply:
Renamed to `Create a new service account that is tied to an AWS IAM role` (being attached to a repo is explained in the repo arg help text and isn't central to the command).